xray/commands/audit/sca/npm/npm.go

```diff
@@ -47,14 +67,78 @@ func BuildDependencyTree(params utils.AuditParams) (dependencyTrees []*xrayUtils
 	return
 }
 
+// We check for the existence of DepsRepo and ServerDetails in the audit params or in a Jfrog config file.
+// If found we configure an artifactory server for resolving dependencies in the current project
+func configureResolutionServerIfNeeded(params utils.AuditParams) (restoreNpmrcFunc func() error, err error) {
+	depsRepo := params.DepsRepo()
+	serverDetails, err := params.ServerDetails()
+	if err != nil {
+		err = fmt.Errorf("couldn't get resolving server details: %s", err.Error())
+		return
+	}
+	if depsRepo == "" || serverDetails == nil {
+		// In case we don't have DepsRepo or ServerDetails, we search for a Jfrog config file, if exists, in order to get them
+		var isNpmConfigFileExists bool
+		npmYamlFilePath := filepath.Join(".jfrog", "projects", "npm.yaml")
+		isNpmConfigFileExists, err = fileutils.IsFileExists(npmYamlFilePath, false)
+		if err != nil {
+			err = fmt.Errorf("failed to check for jfrog config file in the project: %s", err.Error())
+			return
+		}
+		if !isNpmConfigFileExists {
+			return
+		}
+
+		var npmConfigYamlData *viper.Viper
+		npmConfigYamlData, err = utils2.ReadConfigFile(npmYamlFilePath, utils2.YAML)
+		if err != nil {
+			err = fmt.Errorf("couldn't read jfrog NPM configuration file: %s", err.Error())
+			return
+		}
+
+		if serverDetails == nil {
+			serverId := npmConfigYamlData.GetString("resolver.serverId")
+			serverDetails, err = config.GetSpecificConfig(serverId, true, false)
+			if err != nil {
+				err = fmt.Errorf("couldn't get resolving server details: %s", err.Error())
+				return
+			}
+		}
+		if depsRepo == "" {
+			depsRepo = npmConfigYamlData.GetString("resolver.repo")
+		}
+	}
+
+	restoreNpmrcFunc, err = configNpmResolutionServer(depsRepo, serverDetails)
```

Comment:
Is this actually a restore function? What are we restoring here?

Reply:
We restore the state to the previous npmrc (or deleting it if wasn't exist) in order to not leaving any trace to out action after running 'restore' for audit.
This logic is taken from frogbot